docs/tools/offline/mbed-studio.md

```diff
@@ -0,0 +1,193 @@
+
+Mbed Studio automatically switches into the Mbed Debug perspective when debugging with the Mbed Debug launch configuration. The project also breakes on a default break point. The resulting workspace looks like the image below.
+
+##### Advanced debugging
+
+If you want more control over the options regarding debugging, this section is for you. This section describes how to create a PyOCD GDB debug configuration. This configuration allows you to change more settings, such as ports and executable locations for the GDB client and server.
+
+To begin debugging, create the debug launch configuration and set some variables. (You may use your own values if you don’t want to use the default provided here.) To create a debug launch configuration, as before, right-click on a project, and select `Debug As... > Debug Configurations` from the context menu.
+
+A window for the debug configurations appears. Select the `GDB PyOCD Debugging` category, and create a new launch configuration. 
+
+You can click the new icon (a page with a `+` in the upper right corner), or right-click on the category and select `New`. See the screenshot above on the right.
+
+Next, configure the binary you want to use for debugging. This is the `.elf` file located within the project directory and inside `BUILD/<target>/GCC_ARM`. (For this example, it is `BUILD/K64F/GCC_ARM`.) Currently, this is configured manually, so switching to a different target to run the same project requires you to update this field in the debug configuration. See the below left screenshot.
+
+The next step is to configure the location of the executables for the PyOCD GDB Server and the GDB Client on the Debugger tab within the debug configuration. These should be set to the executables located within Mbed Studio’s tools directory which is located at:
+`/eclipse-ws/mbed-studio-ws/tools`
+
+Alternatively, and at your own risk, you may select executables from locations elsewhere on your machine.
+
+On *nix machines, the two bundled executables are located at:
+`/eclipse-ws/mbed-studio-ws/tools/python/bin/pyocd-gdbserver`
+`/eclipse-ws/mbed-studio-ws/tools/gcc-arm/bin/gcc-arm-none-eabi-gdb`
+
+On Windows machines, the two bundled executables are located at:
+`\eclipse-ws\mbed-studio-ws\tools\python\Scripts\pyocd-gdbserver.exe`
+`\eclipse-ws\mbed-studio-ws\tools\gcc-arm\bin\gcc-arm-none-eabi-gdb.exe`
+
+When you have configured everything, select `Apply` in the lower right side of the panel, and the click the blue Debug button just below that.
+
+### Help
+
+This section outlines known issues and provides answers to common questions.
+
+#### Glossary
```

Comment:
Query: Would you prefer we move these terms to the overall Mbed OS glossary? If not, shall we duplicate them there?

Reply:
Fine to move them - as long as we link to them.